DROD/SettingsScreen.cpp

```diff
@@ -1052,6 +1109,86 @@ void CSettingsScreen::OnClick(const UINT dwTagNo)
 #  undef BUTTONTAG_TO_DCMD
 }
 
+//************************************************************************************
+const KeyDefinition *CSettingsScreen::GetOverwrittenModifiableKey(const InputKey newKey, const DCMD eChangedCommand) const
+{
+	const SDL_Keycode newKeycode = ReadInputKey(newKey);
+
+	for (int nCommand = DCMD_NW; nCommand < DCMD_Count; ++nCommand)
+	{
+		if (nCommand == eChangedCommand || !DoesCommandUseModifiers((DCMD)nCommand))
+			continue;
+
+		const KeyDefinition *keyDefinition = GetKeyDefinition(nCommand);
+
+		if (this->pCurrentPlayer->Settings.GetVar(keyDefinition->settingName, UNKNOWN_INPUT_KEY) == (InputKey)newKeycode)
+			return keyDefinition;
+	}
+
+	return NULL;
+}
+
+//************************************************************************************
+void CSettingsScreen::DoKeyRedefinition(const UINT dwTagNo) {
+	if (dwTagNo < TAG_KEY_BUTTON_START || dwTagNo > TAG_KEY_BUTTON_END)
+		return;
+
+	InputKey newKey, currentKey;
+
+	const DCMD eCommand = ButtonTagToDcmd(dwTagNo);
+	const KeyDefinition *keyDefinition = GetKeyDefinition(eCommand);
+	currentKey = this->pCurrentPlayer->Settings.GetVar(keyDefinition->settingName, UNKNOWN_INPUT_KEY);
+
+	if (GetCommandKeyRedefinition(eCommand, currentKey, newKey, !DoesCommandUseModifiers(eCommand)))
+	{
+		if (newKey == currentKey)
+			return;
+
+		// If we are changing a command that does not support modifiers, and provide a key with
+		// modifiers, make sure we don't use the same base key as any of the commands
+		// that have modifiers-variants built-in, eg. ctrl+move
+		if (!DoesCommandUseModifiers(eCommand) && (InputKey)ReadInputKey(newKey) != newKey)
+		{
+			const KeyDefinition *pOverwrittenKey = GetOverwrittenModifiableKey(newKey, eCommand);
+
+			if (pOverwrittenKey) {
+				WSTRING str = L"You can't map to this key because command '";
```

Comment:
This text should be converted into a MID (or two) by adding new entries into a file in the Texts dir (plus entries in DbBase.cpp for patching).